### PR TITLE
expression/agg: add `ResetContext()` for stream aggregation.

### PR DESCRIPTION
**Pull request by winoros**
Benchmark result for `CreateContext` and `ResetContext`:
```
➜  aggregation git:(reset-context) ✗ go test . -bench=BenchmarkCreateContext
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression/aggregation
BenchmarkCreateContext-8   	20000000	        63.1 ns/op	      96 B/op	       1 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression/aggregation	1.388s
```

```
➜  aggregation git:(reset-context) ✗ go test . -bench=BenchmarkResetContext
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression/aggregation
BenchmarkResetContext-8   	300000000	         5.21 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression/aggregation	2.135s
```

```
➜  aggregation git:(reset-context) ✗ go test . -bench=BenchmarkCreateDistinctContext
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression/aggregation
BenchmarkCreateDistinctContext-8   	 1000000	      1171 ns/op	    2912 B/op	       8 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression/aggregation	1.235s
```

```
➜  aggregation git:(reset-context) ✗ go test . -bench=BenchmarkResetDistinctContext
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression/aggregation
BenchmarkResetDistinctContext-8   	 1000000	      1166 ns/op	    2816 B/op	       7 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression/aggregation	1.227s
```

### CODE REVIEW

**Comment by winoros** [comment]
PTAL @zz-jason @XuHuaiyu

**Comment by lamxTyler** [comment]
Update the comment at [here](https://github.com/pingcap/tidb/pull/6097/files#diff-2ed8f699513f49aaaf408561a56aa2e7L350)?

File: expression/aggregation/aggregation.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ type Aggregation interface {
 
 	// Create a new AggEvaluateContext for the aggregation function.
 	CreateContext(sc *stmtctx.StatementContext) *AggEvaluateContext
+
+	ResetContext(sc *stmtctx.StatementContext, evalCtx *AggEvaluateContext)
```
Comment (shenli) [inline expression/aggregation/aggregation.go]: Add a comment for this interface.

**Comment by winoros** [comment]
PTAL @lamxTyler @shenli

**Pull request merged**